### PR TITLE
AG-1169, AG-1170

### PR DESCRIPTION
**Pull request by hallieswan**
- AG-1169: add chiclet for p-value filter
- AG-1170: add url parameter for p-value filter

https://github.com/Sage-Bionetworks/Agora/assets/26949006/a6d8979b-77cc-42dc-a450-a40a8a5ef8bd

https://github.com/Sage-Bionetworks/Agora/assets/26949006/ee256cb7-f151-48c9-bc6d-ba300179d65f

### CODE REVIEW

File: src/app/features/genes/components/gene-comparison-tool/components/gene-comparison-tool-filter-list-item/gene-comparison-tool-filter-list-item.component.html [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<div *ngIf="isVisible" class="gct-filter-list-item">
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/components/gene-comparison-tool-filter-list-item/gene-comparison-tool-filter-list-item.component.html:1]: Refactored the filter list item chiclet into its own component

File: src/app/features/genes/components/gene-comparison-tool/components/gene-comparison-tool-filter-list-item/gene-comparison-tool-filter-list-item.component.scss [hunk 2]
```diff
@@ -0,0 +1,40 @@
+@use 'sass:map';
+
+@import '../../../../../../../styles/variables';
+@import '../../../../../../../styles/mixins';
+
+.gct-filter-list-item {
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/components/gene-comparison-tool-filter-list-item/gene-comparison-tool-filter-list-item.component.scss:6]: These styles are all directly pulled from gct-comparison-tool-filter-list.scss

File: src/app/features/genes/components/gene-comparison-tool/components/gene-comparison-tool-filter-list-item/gene-comparison-tool-filter-list-item.component.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  @Input() item: any;
+  @Input() isVisible = false;
+  @Input() title = '';
+  @Input() description = '';
+  @Output() onClear: EventEmitter<object> = new EventEmitter<object>();
+
+  clearWasClicked() {
+    this.isVisible = false;
+    this.onClear.emit(this.item);
+  }
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/components/gene-comparison-tool-filter-list-item/gene-comparison-tool-filter-list-item.component.ts:25]: Allow the parent to act when the the chiclet remove button is clicked

File: src/app/features/genes/components/gene-comparison-tool/components/gene-comparison-tool-filter-list-item/gene-comparison-tool-filter-list-item.component.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+} from '@angular/core';
+
+@Component({
+  selector: 'gene-comparison-tool-filter-list-item',
+  templateUrl: './gene-comparison-tool-filter-list-item.component.html',
+  styleUrls: ['./gene-comparison-tool-filter-list-item.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class GeneComparisonToolFilterListItemComponent {
+  @Input() item: any;
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/components/gene-comparison-tool-filter-list-item/gene-comparison-tool-filter-list-item.component.ts:16]: The item that will be sent to the parent when the chiclet remove button is clicked, which is used when clearing the GCT filters -- if there's a better way to handle this, happy to adjust!

File: src/app/features/genes/components/gene-comparison-tool/components/gene-comparison-tool-filter-list/gene-comparison-tool-filter-list.component.scss [hunk 5]
```diff
@@ -36,39 +36,3 @@
     }
   }
 }
-
-.gct-filter-list-item {
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/components/gene-comparison-tool-filter-list/gene-comparison-tool-filter-list.component.scss:40]: Moved into gene-comparison-tool-filter-list-item.scss

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.html [hunk 6]
```diff
@@ -129,10 +134,15 @@ <h1 class="gct-heading h2">Gene Comparison Tool</h1>
                         'When on, this will filter out all statistically insignificant results. Current significance cutoff is: ' +
                         significanceThreshold
                       "
+                      styleClass="gct-significance-control-toggle"
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.html:137]: Added this class to more easily find the toggle in tests

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.html [hunk 7]
```diff
@@ -129,10 +134,15 @@ <h1 class="gct-heading h2">Gene Comparison Tool</h1>
...
                       tooltipPosition="left"
                       tooltipStyleClass="tooltip"
                       [(ngModel)]="significanceThresholdActive"
-                      (onChange)="filter()"
+                      (onChange)="
+                        setSignificanceThresholdActive(
+                          significanceThresholdActive
+                        )
+                      "
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.html:145]: Update when the toggle is clicked

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.html [hunk 8]
```diff
@@ -153,11 +163,17 @@ <h1 class="gct-heading h2">Gene Comparison Tool</h1>
                           max="1"
                           step="0.01"
                           [(ngModel)]="significanceThreshold"
+                          (ngModelChange)="
+                            setSignificanceThresholdActive(
+                              significanceThresholdActive
+                            )
+                          "
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.html:170]: Update when the significance threshold value is changed

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.html [hunk 9]
```diff
@@ -153,11 +163,17 @@ <h1 class="gct-heading h2">Gene Comparison Tool</h1>
...
                         />
                       </ng-template>
                     </p-overlayPanel>
                     <button
                       (click)="significanceOverlay.toggle($event)"
+                      class="gct-significance-control-settings"
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.html:176]: Added this class to more easily find the button in tests

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts [hunk 10]
```diff
@@ -138,6 +139,11 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
       this.sortField = this.urlParams.sortField || '';
       this.sortOrder = '1' === this.urlParams.sortOrder ? 1 : -1;
 
+      this.significanceThreshold =
+        this.urlParams.significanceThreshold ||
+        this.DEFAULT_SIGNIFICANCE_THRESHOLD;
+      this.significanceThresholdActive = !!this.urlParams.significanceThreshold;
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts:145]: Set the significance threshold and display the chiclet if the GCT URL includes a significance parameter when the page is loaded

File: src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts [hunk 11]
```diff
@@ -754,6 +770,10 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
       params['pinned'].sort();
     }
 
+    if (this.significanceThresholdActive) {
+      params['significanceThreshold'] = [this.significanceThreshold];
+    }
```
Comment (hallieswan) [inline src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts:775]: Add significance threshold to the URL

**Review by sagely1 — APPROVED** [review]
Looks good!

**Pull request merged**